docs/english/tutorials/random-fact-generator/random-fact-generator.md

```diff
@@ -0,0 +1,297 @@
+
+### Install app in workspace
+
+Still in the app settings, navigate to the **Install App** page in the left sidebar. Install your app. When you press **Allow**, this means you’re agreeing to install your app with the permissions that it’s requesting. Copy the bot token that you receive and store it in a safe place for subsequent steps.
+
+Now that you’ve completed the app setup, it’s time to write some code!
+
+### Install Node and set up the project
+
+Open your terminal or command prompt. First, check to see that you have Node.js installed and it is a recent long-term support (LTS) version by typing the following command.
+
+```sh
+node -v
+```
+
+If you get an error or if it’s an older version than what’s available for download on the [Node.js website](https://nodejs.org/), take a moment to install the latest version.
+
+Next, create a new directory for the project named `useless-fact-app` and change to the new directory.
+
+```sh
+mkdir useless-fact-app
+cd useless-fact-app
+```
+
+For Windows, the command looks like this:
+
+```powershell
+md useless-fact-app
+```
+
+Initialize a new Node.js project using the following command; answer each question with defaults.
+
+```sh
+npm init
+```
+
+Our project is going to use two packages: the Slack [Bolt for JavaScript](/tools/bolt-js/) framework and [Axios](https://www.npmjs.com/package/axios), a popular HTTP client for making API calls. Install these dependencies with the following command.
+
+```sh
+npm install @slack/bolt axios
+```
+
+Open the project in VS Code or your favorite code editor, then create a new file in the project folder named `.env`. Open the file and paste the following, replacing the placeholders with the values you saved earlier. 
+
+```sh
+SLACK_SIGNING_SECRET={your-app-signing-secret}
+SLACK_BOT_TOKEN={your-bot-token-xoxb-1234}
+SLACK_APP_TOKEN={your-app-token-xapp-1234}
+PORT=3000
+```
+
+Next, edit the `package.json` file with the following settings, then save the file.
+* Change `"main"` to:
+
+```json
+"main": "app.js",
+```
+
+* Change `"scripts"` to:
+
+```json
+"scripts": {
+  "dev": "node –env-file=.env –watch app.js"
+}
+```
+
+* Change `"type"` to `"module"` or add this line if it is not already present:
+
+```json
+"type": "module"
+```
+
+### Add function code
+
+Create a new file named `app.js` and add the following code.
+
+```js
+import bolt from "@slack/bolt";
+import axios from "axios";
+
+const { App } = bolt;
+
+// Initialize the Bolt app
+const app = new App({
+  token: process.env.SLACK_BOT_TOKEN,
+  signingSecret: process.env.SLACK_SIGNING_SECRET,
```

Comment:
```suggestion
```

⚡ suggestion: With socket mode as `true` we don't need the signing secret in most cases! OAuth might be an exception?